### PR TITLE
Resto Druid refactor

### PR DESCRIPTION
**Pull request by kfinch**
I'm very sorry about this PR.

The Resto Druid component currently has an enormous amount of disorganized and confusing logic in the CombatLogParser, almost all of which should be in their respective modules. Also a lot of the code uses deprecated functions. I went to move each module's logic into the module's file itself, but the refactor turned out even bigger than I thought; this is the first half.

Note that while I fixed the most in my face uses of deprecated functions, I intentionally did not change code in the deep logic of each module, opting instead to just cut/paste in those cases. The logic will be fixed later, for now I'm attempting to limit the scope of this refactor as much as possible.

I'm sending it in parts because technically the refactor for each module is independent. The resto druid analysis will still look fine if you accept this PR. If you'd prefer to reject this and have me do it all at once, let me know. When the refactor is fully complete I intend to clean up all the commented code in CombatLogParser.

### CODE REVIEW

**Comment by kfinch** [comment]
In case you were wondering, yes I have been in contact with blazyb about this refactor.

**Comment by MartijnHols** [comment]
@buimichael is welcome to review this and merge if it's to his satisfaction. I might not get to reviewing before tomorrow.

**Comment by buimichael** [comment]
@kfinch I rather have the entire refactoring done. I'm not a fan of pushing commented code lines into master. Either remove the comments or/and finish the refactor. I'd like to do some manual testing (in due time) to ensure nothing was broken in this PR.
http://dist.sonarsource.com/plugins/csharp/rulesdoc/0.10.0-RC/S125.html

**Comment by kfinch** [comment]
The Work Continues

**Comment by kfinch** [comment]
There, it's as done as it's ever going to be @buimichael

**Comment by buimichael** [comment]
Thanks! I'd like to take a look into this before it gets merged.

File: src/Parser/RestoDruid/Modules/Legendaries/T20_2Set.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      icon: <SpellIcon id={SPELLS.RESTO_DRUID_T20_2SET_BONUS_BUFF.id} />,
+      title: <SpellLink id={SPELLS.RESTO_DRUID_T20_2SET_BONUS_BUFF.id} />,
+      result: (
+        <dfn data-tip={`This number assumes you used all the extra Swiftmend casts this bonus allowed you, and multiplies by your Swiftmend's average healing. If you weren't good about using Swiftmend on cooldown, the actual benefit to you will be lower than listed.<br>
+        <ul>
+          <li>Gained ${Math.floor(freeSwiftmends)} possible extra Swiftmend casts</li>
+          <li>Refunded an average of ${Math.round(avgCooldownReduction)}s cooldown per Swiftmend cast</li>
+        <ul>
+        `}>
+          {this.owner.formatItemHealingDone(throughput)}
```
Comment (buimichael) [inline src/Parser/RestoDruid/Modules/Legendaries/T20_2Set.js]: I think it's best if we don't translate the extra swiftmends as throughput as there's an opportunity cost of swiftmend (aka they're not free to cast). Instead we should simply show how many free swiftmends was gained. It's a minor thing, either fix it or add it on the to-do list :)

File: src/Parser/RestoDruid/Modules/Legendaries/T21_2Set.js [hunk 2]
```diff
@@ -32,7 +32,6 @@ class T21_2Set extends Module {
       id: `spell-${SPELLS.RESTO_DRUID_T21_2SET_BONUS_BUFF.id}`,
       icon: <SpellIcon id={SPELLS.RESTO_DRUID_T21_2SET_BONUS_BUFF.id} />,
       title: <SpellLink id={SPELLS.RESTO_DRUID_T21_2SET_BONUS_BUFF.id} />,
-      // result: this.owner.formatItemHealingDone(healing),
```
Comment (buimichael) [inline src/Parser/RestoDruid/Modules/Legendaries/T21_2Set.js]: This line should be removed

**Comment by kfinch** [comment]
@buimichael I think you're allowed to merge this at your leisure, or at least that's what @MartijnHols implied in his comment last week.

**Pull request merged**